lambdas/store-identity/src/main/java/uk/gov/di/ipv/core/storeidentity/StoreIdentityHandler.java

```diff
@@ -90,20 +107,32 @@ public Map<String, Object> handleRequest(ProcessRequest input, Context context)
             LogHelper.attachGovukSigninJourneyIdToLogs(
                     clientOAuthSessionItem.getGovukSigninJourneyId());
 
-            verifiableCredentialService.storeIdentity(
+            String userId = clientOAuthSessionItem.getUserId();
+            List<VerifiableCredential> credentials =
                     sessionCredentialsService.getCredentials(
-                            ipvSessionItem.getIpvSessionId(), clientOAuthSessionItem.getUserId()),
-                    clientOAuthSessionItem.getUserId());
+                            ipvSessionItem.getIpvSessionId(), userId);
+            verifiableCredentialService.storeIdentity(credentials, userId);
 
+            if (configService.enabled(EVCS_WRITE_ENABLED)) {
+                var hasF2fVc = credentials.stream().anyMatch(vc -> vc.getCriId().equals(F2F_CRI));
+                CriResponseItem f2fRequest = criResponseService.getFaceToFaceRequest(userId);
+                final boolean isF2FIncomplete = !Objects.isNull(f2fRequest) && !hasF2fVc;
```

Comment:
Do you think this might be better as a `lambdaInput`, so it's controlled by the journey map configuration, rather than messing about with the CRI response service.

I think that'll make it a bit more direct, and less reliant on the exact setup of all the VCs.

Reply:
done